libeos-updater-util/util.c

```diff
@@ -492,3 +493,93 @@ eos_updater_setup_quit_file (const gchar *path,
 
   return g_steal_pointer (&quit_file);
 }
+
+gboolean
+eos_string_to_signed (const gchar  *str,
+                      gint64        min,
+                      gint64        max,
+                      gint64       *out_num,
+                      GError      **error)
+{
+  gint64 number;
+  const gchar *end_ptr = NULL;
+
+  g_return_val_if_fail (str != NULL, FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
```

Comment:
Note to self: add `g_return_val_if_fail (min <= max, FALSE);` here and in the unsigned variant.